DeviceAdministration/Infrastructure.UnitTests/Infrastructure/IoTHubRepositoryTests.cs

```diff
@@ -0,0 +1,134 @@
+            Assert.True(result);
+
+            result = await this.iotHubRepository.TryAddDeviceAsync(null);
+            Assert.False(result);
+        }
+
+        [Fact]
+        public async void GetIotHubDeviceAsync()
+        {
+            var deviceId = this.fixture.Create<string>();
+            this.deviceManagerMock.Setup(dm => dm.GetDeviceAsync(deviceId))
+                .Returns(Task.FromResult(new Device(deviceId)));
+            var d = await this.iotHubRepository.GetIotHubDeviceAsync(deviceId);
+            Assert.Equal(deviceId, d.Id);
+        }
+
+        [Fact]
+        public async void RemoveDeviceAsync()
+        {
+            var deviceId = this.fixture.Create<string>();
+            await this.iotHubRepository.RemoveDeviceAsync(deviceId);
+        }
+
+        [Fact]
+        public async void TryRemoveDeviceAsync()
+        {
+            var deviceId = this.fixture.Create<string>();
+            var result = await this.iotHubRepository.TryRemoveDeviceAsync(deviceId);
+            Assert.True(result);
+
+            this.deviceManagerMock.Setup(dm => dm.RemoveDeviceAsync(It.IsAny<string>())).Throws(new Exception());
+            result = await this.iotHubRepository.TryRemoveDeviceAsync(null);
+            Assert.False(result);
+        }
+
+        [Fact]
+        public async void UpdateDeviceEnabledStatusAsync()
+        {
+            var deviceId = this.fixture.Create<string>();
+            var device = new Device(deviceId);
+            device.Status = DeviceStatus.Enabled;
+
+            this.deviceManagerMock.Setup(dm => dm.GetDeviceAsync(deviceId))
+                .Returns(Task.FromResult(device));
+
+            this.deviceManagerMock.Setup(dm => dm.UpdateDeviceAsync(It.IsAny<Device>()))
+                .Returns(Task.FromResult(device));
+
+            var sameDevice = await this.iotHubRepository.UpdateDeviceEnabledStatusAsync(deviceId, false);
+            Assert.Equal(sameDevice.Status, DeviceStatus.Disabled);
+
+            sameDevice = await this.iotHubRepository.UpdateDeviceEnabledStatusAsync(deviceId, true);
+            Assert.Equal(sameDevice.Status, DeviceStatus.Enabled);
+        }
+
+        [Fact]
+        public async void SendCommand()
+        {
+            var commandHistory = this.fixture.Create<CommandHistory>();
+            var deviceId = this.fixture.Create<string>();
+            this.deviceManagerMock.Setup(dm => dm.SendAsync(deviceId, It.IsAny<Message>())).Returns(Task.FromResult(true));
+            this.deviceManagerMock.Setup(dm => dm.CloseAsyncDevice()).Returns(Task.FromResult(true));
+
+            await this.iotHubRepository.SendCommand(deviceId, commandHistory);
+        }
+
+        [Fact]
+        public async void GetDeviceKeysAsync()
+        {
+            var deviceId = this.fixture.Create<string>();
+            var device = new Device(deviceId);
+            var auth = new AuthenticationMechanism();
+            auth.SymmetricKey = new SymmetricKey();
```

Comment:
previous comment. It has to be in a specific format so I'm just initializing them myself.